pkg/log/log_file.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main under directory cmd parses and validates user input,
+// instantiates and initializes objects imported from pkg, and runs
+// the process.
+package log
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+const (
+	logToStdErrFlag = "logtostderr"
+	logDirFlag      = "log_dir"
+	logFileFlag     = "log_file"
+	maxSizeFlag     = "log_file_max_size"
+	maxNumFlag      = "log_file_max_num"
+
+	// Check log file number every 10 mins.
+	logFileCheckInterval = time.Minute * 10
+)
+
+var (
+	maxNumArg     = uint16(0)
+	logFileMaxNum = uint16(0)
+	logDir        = ""
+
+	executableName = filepath.Base(os.Args[0])
+)
+
+func AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&maxNumArg, maxNumFlag, maxNumArg, "Maximum number of log files per severity level to be kept. Value 0 means unlimited.")
+}
+
+// InitLogFileLimits initializes log file maximum size and maximum number limits based on the
+// command line flags.
+func InitLogFileLimits(fs *pflag.FlagSet) {
+	var err error
+	var logToStdErr bool
+	var logFile string
+	var maxSize uint64
+
+	logToStdErr, err = fs.GetBool(logToStdErrFlag)
+	if err != nil {
+		// Should not happen. Return for safety.
+		return
+	}
+	if logToStdErr {
+		// Logging to files is not enabled.
+		return
+	}
+
+	logFile, err = fs.GetString(logFileFlag)
+	if err != nil {
+		return
+	}
+	if logFile != "" {
+		// Log to a single file. klog will take care of the max size limit.
+		return
+	}
+
+	// Max log file size in MB.
+	maxSize, err = fs.GetUint64(maxSizeFlag)
+	if err != nil {
+		return
+	}
+	if maxSize > 1024*1024 {
+		klog.Errorf("The specified log file max size %d is too big, ignored", maxSize)
```

Comment:
maybe quantify "too big" ? with a number that we allow

Reply:
Ok.